lib/SecretsPluginApi/extensionplugins.cpp

```diff
@@ -13,71 +13,1189 @@
+ * Sailfish::Secrets::Result::OperationNotSupportedError).
+ *
+ * If no collection with the given \a collectionName exists in the storage
+ * managed by the plugin, the plugin should return a Sailfish::Secrets::Result
+ * with the result code set to Sailfish::Secrets::Result::Failed and the
+ * error code set to Sailfish::Secrets::Result::InvalidCollectionError.
+ *
+ * If a secret with the specified \a secretName is already stored
+ * in the collection identified by the given \a collectionName, the plugin
+ * should return a Sailfish::Secrets::Result with the result code set to
+ * Sailfish::Secrets::Result::Failed and the error code set to
+ * Sailfish::Secrets::Result::SecretAlreadyExistsError.
+ */
+
+/*!
+ * \fn StoragePlugin::getSecret(const QString &collectionName, const QString &secretName, QByteArray *secret, Sailfish::Secrets::Secret::FilterData *filterData)
+ * \brief Write the secret data and filter data associated with the secret
+ *        identified by the given \a secretName in the collection identified
+ *        by the given \a collectionName into the \a secret and \a filterData
+ *        out-parameters respectively.
+ *
+ * If the plugin itself is locked, this function should return a
+ * Sailfish::Secrets::Result with the result code set to
+ * Sailfish::Secrets::Result::Failed and the error code set to
+ * Sailfish::Secrets::Result::SecretsPluginIsLockedError.
+ *
+ * If no collection with the given \a collectionName exists in the storage
+ * managed by the plugin, the plugin should return a Sailfish::Secrets::Result
+ * with the result code set to Sailfish::Secrets::Result::Failed and the
+ * error code set to Sailfish::Secrets::Result::InvalidCollectionError.
+ *
+ * If no secret with the given \a secretName exists in that collection
+ * managed by the plugin, the plugin should return a Sailfish::Secrets::Result
+ * with the result code set to Sailfish::Secrets::Result::Failed and the
+ * error code set to Sailfish::Secrets::Result::InvalidSecretError.
+ *
+ * If the secret data was retrieved successfully, the plugin should return a
+ * Sailfish::Secrets::Result with the result code set to
+ * Sailfish::Secrets::Result::Succeeded.
+ */
+
+/*!
+ * \fn StoragePlugin::secretNames(const QString &collectionName, QStringList *secretNames)
+ * \brief Write the names of secrets which are stored by the plugin in the
+ *        collection with the given \a collectionName to the \a secretNames
+ *        out-parameter.
+ *
+ * If the plugin itself is locked, this function should return a
+ * Sailfish::Secrets::Result with the result code set to
+ * Sailfish::Secrets::Result::Failed and the error code set to
+ * Sailfish::Secrets::Result::SecretsPluginIsLockedError.
+ *
+ * If the given \a collectionName is empty, or is the special value
+ * "standalone", the plugin should return the names of standalone secrets
+ * which are stored in the storage managed by the storage plugin.
```

Comment:
What should be the return value if collectionName is empty or standalone and the plugin does not support standalone secrets?

Reply:
Clarified that it should return success after setting the outparam to an empty list.